### PR TITLE
Updating to reflect changes since OH 2 pre release

### PR DESCRIPTION
**Pull request by rkoshak**
Adopted the one sentence per line convention used everywhere else in the docs.

Rewrote the paragraph discussing the versions of OH 2 available (Release, Testing, SNAPSHOT). NOTE: the Installation Overview needs to be updated as well. 

Removed discussion of offline and online since I don't think that is correct anymore. There is no mention of different offline/online installs in the installation docs.

Tried to make it more clear that the text based instructions are the recommended.

Replaces 1.9.0 with 1.x.0 and replaced 2.0 with 2.x to make the instructions not specific to the version.

Removed the warnings to create the persistence folder when running in Docker. They are no longer necessary.

In necessary changes for Rules, I removed the "core" from the org.openhab imports as there are some users who might be using script imports as well.

Removed references to ESHD and changed them to reference VSCode instead. Solves #775 

Signed off by Richard Koshak rlkoshak@gmail.com

### CODE REVIEW

File: tutorials/migration.md [hunk 1]
```diff
@@ -22,178 +25,122 @@ This page is structured as follows:
...
-Machine, changes to the default ports, etc.
+- `/var/lib/openhab` - openhab user's home directory, zwave and embedded persistence database files
+- `/usr/share/openhab/webapps` - custom icons, custom webviews (e.g. [Weather Binding](https://github.com/openhab/openhab/wiki/Weather-Binding))
+- `/usr/share/openhab/bin` - custom command line arguments to the Java Virtual Machine, changes to the default ports, etc.
 - `/etc/default/openhab` - an alternative place to change default ports
 
-If you are on an operating system that does not support apt-get or performed a
-manual installation for some other reason simply backup the whole
-`<openhab home>` directory.
+If you are on an operating system that does not support apt-get or performed a manual installation for some other reason simply backup the whole `<openhab home>` directory.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.
See https://github.com/openhab/openhab-docs/issues/189#issuecomment-360202149 for an overview and a link where to find the source.

So in this case `$OPENHAB_HOME` would be the right choice.

File: tutorials/migration.md [hunk 2]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
-add-on.
+When you find a binding to install, press the Install button and wait for installation to complete. 
+Do not install the myopenhab/openhabcloud Binding yet.
 
-One thing to be aware of is any configuration done in PaperUI or Habmin gets
-saved to a non-human readable database. And since none of the administration
-UIs are currently capable of doing everything, you will end up with a mix of
-database and text based configuration. This is why we recommend not using
-PaperUI for those migrating to openHAB 2 from openHAB 1.x.
+One thing to be aware of is any configuration done in PaperUI gets saved to a files and folders under <openHAB 2 conf>.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 3]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
+Binding configurations, Items, Things, et. al. created in, edited in, and otherwise modified in PaperUI will not appear in the conf directories.
+For example, if you create an Item in PaperUI, you will not find that new Item in any of the .items files.
+Instead they get saved to a JSONDB file located in <openHAB 2 userdata>/jsondb.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 4]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
+Errors at this time may not be important but make note of those that did generate errors during installation for special attention later.
 
-Once a binding or add-on is installed, it will create a
-`<openHAB 2 conf>/services/<add-on name>.cfg` file. Unlike in openHAB 1.x where all the
-binding configurations are placed in the one openhab.cfg file, openHAB 2 has a
-separate .cfg file for each binding. Transfer the settings for each binding from
-openhab.cfg to its new .cfg file, removing the binding name from the parameter.
-For example, the parameter `nest:refresh=300000` in openhab.cfg becomes
-`refresh=300000` in nest.cfg.
+Once a binding or add-on is installed, it will create a `<openHAB 2 conf>/services/<add-on name>.cfg` file.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 5]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
-Copy your openhab.cfg file to `<openHAB 2 conf>/services`. If you are running an apt-get
-installed openHAB 1.x openhab.cfg is located in `/etc/openhab/configurations`.
-For an apt-get installed openHAB 2 the `<openHAB 2 conf>` folder is located in
-`/etc/openhab2`. For manually installed openHAB 1.x this file is located
-in `<openhab home>/configurations` and in manually installed openHAB 2 it is
-`<openhab home>/conf`.
+Copy your openhab.cfg file to `<openHAB 2 conf>/services`. 
+If you are running an apt-get installed openHAB 1.x openhab.cfg is located in `/etc/openhab/configurations`.
+For an apt-get installed openHAB 2 the `<openHAB 2 conf>` folder is located in `/etc/openhab2`. 
+For manually installed openHAB 1.x this file is located in `<openhab home>/configurations` and in manually installed openHAB 2 it is `<openhab home>/conf`.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 6]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
+These instructions are only for add-ons that are not listed in PaperUI or the Karaf Console.
 
 ### Final Add-ons Installation Steps
 
-openHAB 2 has a different
-[folder layout]({{base}}/installation/linux.html#file-locations).
-Of particular node are the configuration folders and userdata folders. See the
-previous link for the location of these folders for your installtion.
+openHAB 2 has a different [folder layout]({{base}}/installation/linux.html#file-locations).
+Of particular node are the <openHAB 2 conf> folders and <openHAB 2 userdata> folders.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 7]
```diff
@@ -796,21 +679,18 @@ becomes
...
-All definitions of new Things are written to .things files in the
-`<openHAB 2 conf>/things` folder. The syntax for a Thing definition varies from
-binding to binding. See the binding's readme for the specific format and parameters
-required. Also see the binding's readme for the list of channels the Thing
-supports. As discussed above, the Channel ID will be the `<Thing ID>#<channel>`.
+Not all 2.0 bindings support automatic discovery of Things or by their very nature require manual creation of Things. 
+As with linking Channels to Items, there is more than one way to do this including through PaperUI, Habmin, the REST API, or through text files. 
+This tutorial will only cover creating Things in text files as that will be closer to the openHAB 1.x way of doing things.
+
+All definitions of new Things are written to .things files in the `<openHAB 2 conf>/things` folder.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 8]
```diff
@@ -642,87 +543,73 @@ Back up your openHAB 1.x configurations and uninstall openHAB 1.x if desired.
...
 
-Using your preferred method for add-on installation (see above) install the
-My.openHAB binding. Once installed a new `uuid` and `secret` file will be created.
-These two files are located in a different place in openHAB 2:
-`<openhab 2 userdata>/uuid` and `<openHAB 2 userdata>/myopenhab/secret`.
-As of this writing, the http://my.openhab.org documentation still references the
-openHAB 1.x locations.
+Using your preferred method for add-on installation (see above) install the openHAB Cloud Connector binding (Misc tab in PaperUI). 
+Once installed a new `uuid` and `secret` file will be created.
+These two files are located in a different place in openHAB 2: `<openhab 2 userdata>/uuid` and `<openHAB 2 userdata>/myopenhab/secret`.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 9]
```diff
@@ -463,78 +391,71 @@ cp <openHAB 1.x conf>/configurations/sitemaps/* <openHAB 2 conf>/sitemaps/*
...
-Continue watching the logs as you move files over. There will likely be a number
-of errors. Take note of them with plans to return and correct them if they
-persist.
+Continue watching the logs as you move files over. 
+There will likely be a number of errors. 
+Take note of them with plans to return and correct them if they persist.
 
 ## Testing
 
 - **Items**: Watch `<openHAB 2 userdata>/logs/events.log` and methodically work through all
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 10]
```diff
@@ -463,78 +391,71 @@ cp <openHAB 1.x conf>/configurations/sitemaps/* <openHAB 2 conf>/sitemaps/*
 
   * Static webview files are now located in `<openhab 2 conf>/html` instead of  `/usr/share/openhab/webapps`
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

File: tutorials/migration.md [hunk 11]
```diff
@@ -292,134 +237,112 @@ To get a list of all currently installed items run:
...
-This file will have a lot of redundant configuration information in it that you
-have already moved over to individual binding .cfgs. Make sure to comment those
-out and just leave the config parameters for the bindings you are manually
-installing.
+This file will have a lot of redundant configuration information in it that you have already moved over to individual binding .cfgs. 
+Make sure to comment those out and just leave the config parameters for the bindings you are manually installing.
 
-Now copy the add-on's jar files that you want to install to the `<openHAB conf>/addons` folder for
-openHAB 2. For example:
+Now copy the add-on's jar files that you want to install to the `<openHAB conf>/addons` folder for openHAB 2.
```
Comment (Confectrician) [inline tutorials/migration.md]: According to #189 we should use a consistent `$FOLDER` notation for all articles.

**Comment by rkoshak** [comment]
I think I got all the instances. It should now comply with #189 now.

**Pull request merged**